### PR TITLE
[FIX] ci: Introduced new matrix syntax and removed unnecessary tox envs

### PR DESCRIPTION
**Pull request by antonag32**
Similar REF done to odoo-pre-commit-hooks and pylint-odoo

### CODE REVIEW

**Comment by antonag32** [comment]
@moylop260 CI should go green now (although some tests seem to be stuck, queued...), besides the refactoring I also had to fix some issues in the source code.

File: .github/workflows/github-actions.yml [hunk 1]
```diff
@@ -29,121 +29,27 @@ jobs:
...
-            python: '3.11-dev'
-            toxpython: 'python3.11'
-            python_arch: 'x64'
-            tox_env: 'py311,report,codecov'
-            os: 'macos-latest'
-
+          - python: '3.10'
+            os: ubuntu-latest
+            tox_env: 'docs'
+          - python: '3.10'
```
Comment (moylop260) [inline .github/workflows/github-actions.yml:47]: what about update-readme?

(in order to check if there is a diff)

Comment (antonag32) [reply .github/workflows/github-actions.yml:47]: It was commented out in `tox.ini` and I could not find any reference to it in github-actions so I don't think it was running previously and removed it, I can enable it if you like to.

Comment (moylop260) [reply .github/workflows/github-actions.yml:47]: OMG

You just find another missing thing

Yes, I think it is needed

Do the another projects has the same situation?

Comment (antonag32) [reply .github/workflows/github-actions.yml:47]: `pylint-odoo` and `odoo-pre-commit-hooks` do run `update-readme` so no issues there.

I was looking into it and it seems that there is no test_build_docstring in the source code, so we will need to write it before adding it to the CI. 

This is the commit where `update-readme` was added to `tox.ini`: [39156804](https://github.com/Vauxoo/pre-commit-vauxoo/commit/39156804#diff-8b66610a43eb066fd4bbb4d99ad34425ffaec81be10d0174e86252b7a09dbe34)

It seems we forgot to add the test for it.

**Pull request merged**